app/src/common/shared/com/igalia/wolvic/ui/adapters/WebApp.java

```diff
@@ -130,19 +125,19 @@ public boolean equals(Object o) {
             return false;
         WebApp webApp = (WebApp) o;
         return mIdentity.equals(webApp.mIdentity) &&
-                mName.equals(webApp.mName) &&
-                mShortName.equals(webApp.mShortName) &&
-                mScope.equals(webApp.mScope) &&
-                mStartUrl.equals(webApp.mStartUrl) &&
-                mThemeColor == webApp.mThemeColor &&
-                mBackgroundColor == webApp.mBackgroundColor;
+                getName().equals(webApp.getName()) &&
+                getShortName().equals(webApp.getShortName()) &&
+                getScope().equals(webApp.getScope()) &&
+                getStartUrl().equals(webApp.getStartUrl()) &&
+                getThemeColor() == webApp.getThemeColor() &&
+                getBackgroundColor() == webApp.getBackgroundColor();
```

Comment:
Doesn't WebApp implement the `equals()` method?

Reply:
You're right, `WebAppManifest` is a "data class" and those get `equals()` for free in Kotlin: https://kotlinlang.org/docs/data-classes.html